spec/lang/representation.md

```diff
@@ -155,17 +156,19 @@ impl PtrType {
     }
 }
 
-impl PointerMeta {
-    fn from_value<M: Memory>(value: Value<M>) -> Option<PointerMeta> {
+impl<Provenance> PointerMeta<Provenance> {
+    fn from_value<M: Memory<Provenance=Provenance>>(value: Value<M>) -> Option<PointerMeta<Provenance>> {
         match value {
             Value::Int(count) => Some(PointerMeta::ElementCount(count)),
+            Value::Ptr(ptr) => Some(PointerMeta::VTablePointer(ptr.thin_pointer)),
```

Comment:
Can you ensure that `ptr` here does not have metadata? That should return `None`.

Or should it panic? Can this ever be called with non-metadata values?

Reply:
If callers always have the `PointerMetaKind` available, then that seems cleaner, yes. Can you adjust this?